File: backend/src/main/scala/domains/auth/model/UserPreferences.scala
package domains.auth.model

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UserPreferences(
  displayMode: UserDisplayMode
)

object UserPreferences:
  given Encoder[UserPreferences] = deriveEncoder[UserPreferences]
  given Decoder[UserPreferences] = deriveDecoder[UserPreferences]

File: frontend/src/features/auth/model/UserPreferences.ts
import type { UserDisplayMode } from '@/features/auth/model/UserDisplayMode'
import type { UserLocale } from '@/features/auth/model/UserLocale'
import type { ProblemTitleDisplayMode } from '@/features/problem/model/ProblemTitleDisplayMode'

export type UserPreferences = {
  displayMode: UserDisplayMode
  locale: UserLocale
  problemTitleDisplayMode: ProblemTitleDisplayMode
}
